src/common/src/util/value_encoding/mod.rs

```diff
@@ -225,6 +233,28 @@ fn serialize_scalar(value: ScalarRefImpl<'_>, buf: &mut impl BufMut) {
     }
 }
 
+fn estimate_serialize_scalar_size(value: ScalarRefImpl<'_>) -> usize {
+    match value {
+        ScalarRefImpl::Int16(_) => 2,
+        ScalarRefImpl::Int32(_) => 4,
+        ScalarRefImpl::Int64(_) => 8,
+        ScalarRefImpl::Serial(_) => 8,
+        ScalarRefImpl::Float32(_) => 4,
+        ScalarRefImpl::Float64(_) => 8,
+        ScalarRefImpl::Utf8(v) => estimate_serialize_str_size(v.as_bytes()),
+        ScalarRefImpl::Bytea(v) => estimate_serialize_str_size(v),
+        ScalarRefImpl::Bool(_) => 1,
+        ScalarRefImpl::Decimal(_) => estimate_serialize_decimal_size(),
+        ScalarRefImpl::Interval(_) => estimate_serialize_interval_size(),
+        ScalarRefImpl::NaiveDate(_) => estimate_serialize_naivedate_size(),
+        ScalarRefImpl::NaiveDateTime(_) => estimate_serialize_naivedatetime_size(),
+        ScalarRefImpl::NaiveTime(_) => estimate_serialize_naivetime_size(),
+        ScalarRefImpl::Jsonb(_) => 8,
```

Comment:
Is this a mistake? @st1page

Reply:
It’s heavy to pre-compute the size of jsonb, but anyway, 8 is too small and we need some comments here.

Reply:
🤔 I will leave an issue to bench and trade-off if it is worth estimating the jsonb's size. @st1page